File: src/main/java/me/elsiff/morefish/condition/ThunderingCondition.java
package me.elsiff.morefish.condition;

import org.bukkit.entity.Player;

public class ThunderingCondition implements Condition {
    private final boolean thundering;

    public ThunderingCondition(boolean thundering) {
        this.thundering = thundering;
    }

    @Override
    public boolean isSatisfying(Player player) {
        return (thundering == player.getWorld().isThundering());
    }
}

File: src/main/kotlin/me/elsiff/morefish/fishing/condition/ThunderingCondition.kt
package me.elsiff.morefish.fishing.condition

import me.elsiff.morefish.fishing.competition.FishingCompetition
import org.bukkit.entity.Item
import org.bukkit.entity.Player

/**
 * Created by elsiff on 2019-01-12.
 */
class ThunderingCondition(
    private val thundering: Boolean
) : FishCondition {
    override fun check(
        caught: Item,
        fisher: Player,
        competition: FishingCompetition
    ): Boolean {
        return caught.world.isThundering == thundering
    }
}
